core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/DetailedSampleImpl.java

```diff
@@ -69,6 +69,8 @@
 
   @Transient
   private Long identityId;
+  @Transient
+  private String externalNames;
```

Comment:
Rather than having this transient field that will usually be null, can you set `externalNames` on the Identity during conversion from DTO? Then in the service, you can compare the submitted Sample's Identity's `externalNames` to what it has in the DB to determine whether it needs updated.